src/ivc/incrementally_verifiable_computation.rs

```diff
@@ -71,19 +79,233 @@ where
+    ///    secondary contexts. This involves defining the layout and constraints of the circuits
+    ///    based on the `StepCircuit` trait implementations for `SC1` and `SC2`.
+    ///
+    /// 3. **Initial State Assignment**: The initial states (`z0_primary` and `z0_secondary`) are
+    ///    assigned to the constraint system. This forms the base case for the IVC scheme, representing
+    ///    the starting point of the computation.
+    ///
+    /// 4. **Synthesizing the Primary Step Circuit**: Constructs the primary step circuit by
+    ///    applying constraints and recording output states (`z_i`), encoding the computation for
+    ///    the zero round. After run The Sumcheck Protocol over Polynomial Systems (SPS Protocol) for
+    ///    receive generates [`PlonkInstance`] (`u`) & [`crate::plonk::PlonkWitness`]
+    ///
+    /// 5. **Synthesizing the Secondary Step Circuit**: Similarly, constructs the secondary step
+    ///    circuit. Notably, it passes the [`PlonkInstance`] from the primary circuit synthesis as
+    ///    [`StepInputs::u`].After run The Sumcheck Protocol over Polynomial Systems (SPS Protocol) for
+    ///    receive generates [`PlonkInstance`] (`u`) & [`crate::plonk::PlonkWitness`]
+    ///
+    /// 6. **Creating StepCircuitContext**: For both primary and secondary circuits, a
+    ///    [`StepCircuitContext`] is created, encapsulating the step circuit, its relaxed trace, and
+    ///    the initial and final states.
+    ///
+    /// 7. **Finalizing IVC Instance**: The function concludes by constructing and returning the
+    ///    `IVC` instance, which holds the contexts for both circuits.
     pub fn new<const T: usize, RP1, RP2>(
-        _pp: &PublicParams<'_, A1, A2, T, C1, C2, SC1, SC2, RP1, RP2>,
-        _primary: SC1,
-        _z0_primary: [C1::Scalar; A1],
-        _secondary: SC2,
-        _z0_secondary: [C2::Scalar; A2],
+        pp: &PublicParams<'_, A1, A2, T, C1, C2, SC1, SC2, RP1, RP2>,
+        primary: SC1,
+        z0_primary: [C1::Scalar; A1],
+        secondary: SC2,
+        z0_secondary: [C2::Scalar; A2],
     ) -> Result<Self, Error>
     where
         RP1: ROPair<C1::Scalar, Config = MainGateConfig<T>>,
         RP2: ROPair<C2::Scalar, Config = MainGateConfig<T>>,
     {
         // TODO #31
-        todo!("Logic at `RecursiveSNARK::new`")
+        info!("start ivc base case");
+
+        let primary_public_params_hash = pp.digest()?;
+        let secondary_public_params_hash = pp.digest()?;
+
+        let (primary_config, mut primary_td) =
+            pp.primary.prepare_td(&[C1::Scalar::ZERO, C1::Scalar::ZERO]);
+        debug!("primary step circuit configured");
+
+        let (secondary_config, mut secondary_td) = pp
+            .secondary
+            .prepare_td(&[C2::Scalar::ZERO, C2::Scalar::ZERO]);
+        debug!("secondary step circuit configured");
+
+        let primary_cross_term_commits_len = secondary_td
+            .plonk_structure()
```

Comment:
PlonkStructure does not account for on-circuit-time parts in any way? Like copy-constraints? They are calculated at the synthesis stage

Reply:
That is, I eval plonk structure three times in this function (inside digest, for calculating the commitment dimension and for the final traces), but I did so only because I thought that plonk structure is affected by synthetize, namely lookup table is built on the basis of synthetize

Reply:
PlonkStructure doesn't affect the on-circuit-time, only affect the off-circuit computation time.  Currently, the synthesize of TableData will get all the data we needed and then `plonk_structure()` will return the PlonkStructure, which we can reuse. 
After we separate synthesize of TableData into two phases in #146, it will be more efficient.

Reply:
> IIRC, If the degree is `d`, the number of cross term is `d-1`, then the length is `d-1` instead.

only having this question left

Reply:
You didn't understand my question.

During synthesis, lookup arguments are created. This is the only non witness parameter that is created during synthesis. 

As far as I understand, the plonk structure should therefore be different BEFORE and AFTER synthesizing `StepCircuit`, because AFTER it will include the copy constraint.

Actually there is no problem to calculate `PlonkStructure` once, my only question is whether it is correct.

Reply:
Degree fixed

Reply:
You are right that the plonk_structure will be different before and after synthesize of step circuit. Not just copy constraint, it might add extra custom gates and lookup arguments as well. To have all information, we need call `plonk_structure()` after synthesize step circuit. So if we need call it before step circuit synthesize, then we have to pay for the cost of call it multiple times.

Reply:
Well, custom gates can't be set at the synthesis stage, or am I missing something?

Otherwise, any comments on PR?

Reply:
`assembly()` inside call `td.configure` & we always call it for our `td` 

I have dedicated a separate method for this purpose (it used to be called `prepare_assembly`)